app/controllers/api/v2/registration_controller.rb

```diff
@@ -117,7 +119,7 @@ def prepare_host
         host_attributes = @host.apply_inherited_attributes(host_params('host'))
 
         @host.assign_attributes(host_attributes)
-        @host.owner = User.current
+        @host.owner = User.find(params[:host][:owner_id]) || User.current
```

Comment:
Everywhere else `host_params` is used. What's the difference? I'd also expect that if it's in params already that it would be set via `host_attributes`. Perhaps this code is setting it twice now.